### PR TITLE
[T5A2][W13-B4] Conan Kian

### PR DESCRIPTION
**Pull request by ckjr**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -25,6 +25,12 @@ public Command(int targetIndex) {
 
     protected Command() {
     }
+    
+    /**
+     * This method will return true for command types that mutate the data. e.g. AddCommand
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/Command.java]: first, please follow the format that is specified for header comments.
next, try to be more specific on this header comment; the abstract class is a contract that we are providing to the client classes, hence the documentation should be precise. e.g., think what happens when you try to add a duplicate entry to address book. does it still mutate data?

File: src/seedu/addressbook/logic/Logic.java [hunk 2]
```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if (command.isMutating()) {
+            storage.save(addressBook);
```
Comment (okkhoy) [inline src/seedu/addressbook/logic/Logic.java]: may be you need to check how `LogicTest` works now? does it fail?

**Comment by okkhoy** [comment]
@ckjr 
Some comments added. Please ack & close the PR after reading comments.

**Comment by ckjr** [comment]
Acknowledged

**Pull request closed without merging**